docs/email/postfix/postfix-spf-dkim-debian-8.md

```diff
@@ -0,0 +1,376 @@
+
+To add the record, go to your DNS management interface and add a record of type TXT for your domain itself (ie. a blank hostname) containing this string:
+
+    "v=spf1 mx -all"
+
+If you're using Linode's DNS Manager, go to the domain zone page for the domain you want to set SPF up for and add a new TXT record. The screen will look something like this when you've got it filled out:
+
+![Linode DNS manager add TXT record](/docs/assets/Postfix_SPF_TXT_record.png)
+
+If your DNS provider allows it (DNS Manager doesn't) you should also add a record of type SPF, filling it in the same way as you did the TXT record.
+
+### Adding the SPF policy agent to Postfix
+
+The Python SPF policy agent adds SPF policy checking to Postfix. The SPF record for the sender's domain for incoming mail will be checked and if it exists mail will be handled accordingly. There's a Perl version of the policy agent, but it lacks the full capabilities of the Python one.
+
+1.  If you are using SpamAssassin to do spam filtering, you may want to edit `/etc/postfix-policyd-spf-python/policyd-spf.conf` to change the `HELO_reject` and `Mail_From_reject` settings to `False`. That will cause the SPF policy agent to run it's tests and and add a message header with the results in it but _not_ reject any messages. You may also want to make this change if you want to see the results of the checks but not actual apply them to mail processing. Otherwise, just go with the standard settings.
+
+2.  Edit `/etc/postfix/master.cf` and add the following entry at the end:
+
+{: .file-excerpt}
+/etc/postfix/master.cf
+:   ~~~ text
+    policyd-spf  unix  -       n       n       -       0       spawn
+        user=policyd-spf argv=/usr/bin/policyd-spf
+    ~~~
+
+3.  Edit `/etc/postfix/main.cf` as follows:
+
+    1.  Add this entry to increase the timeout Postfix uses with the policy agent to prevent Postfix from aborting the agent if transactions are running a bit slowly:
+
+{: .file-excerpt}
+/etc/postfix/main.cf
+:   ~~~ conf
+    policy_time_limit = 3600
+    ~~~
+
+    2.  Edit the `smtpd_recipient_restrictions` entry to add a `check_policy_service` entry thusly:
```

Comment:
The smtpd_recipient_restrictions entry? It won't if you haven't completed configuring Postfix for normal Internet operation without being an open relay (that's why the second note, I'm converting that to a caution to emphasize it).